VerilogForVerificationStyle.md

```diff
@@ -0,0 +1,863 @@
+```
+
+7.  All type or instance overrides must be in place before creating the class instance.
+    Factory usage examples:
+
+```systemverilog
+// Override all drivers in testbench with my_driver
+factory.set_type_override_by_type(current_driver::get_type(), my_driver::get_type());
+
+// Override all drivers in testbench with my_driver - alternative syntax
+current_driver::type_id::set_type_override(my_driver::get_type());
+
+// Override specific instance of current_driver with my_driver
+factory.set_inst_override_by_type("env1.m_agent1.driver", current_driver::get_type(), my_driver::get_type());
+
+// Override specific instance of current driver - alternative syntax
+current_driver::type_id::set_inst_override(my_driver::get_type(), "env.m_agent1.driver");
+```
+
+
+### Configuration Mechanism
+
+When using the UVM configuration database, these guidelines should be followed:
+
+1.  The `uvm_config_db` API must be used. Do not use `uvm_resource_db`. Do not use
+    `set/get_config_object()`, `set/get_config_string()`, or `set/get_config_int()`,
+    these have been deprecated.
+2.  In general, a designated configuration object extended from `uvm_object`
+    should be used for every `uvm_env` and `uvm_agent` component in the testbench.
+3.  It is recommended that the environment's configuration object contain the
+    agent's configuration object.
+4.  The `uvm_config_db` API should be used to pass configuration objects to
+    locations where they are needed. It should not be used to pass integers,
+    strings, or other basic data types, since it is much easier for namespace
+    collisions to occur when using lower level types.
+5.  You must always check for success when accessing the `uvm_config_db`.
+    If the lookup fails, issue a `uvm_fatal` to terminate the simulation.
+    This should be done as follows:
+
+```systemverilog
+if (!uvm_config_db#(...)::get(...)) begin
+  `uvm_fatal(...)
+end
+```
+6.  Wildcards must not be used in the field argument of `uvm_config_db::set(...)` calls.
+    It is allowed to use wildcards in the inst_name argument of these calls,
+    with proper precautions.
+
+
+### Sequences
+
+1.  It is recommended to keep sequences as generic as possible and to avoid
+    writing directed sequences unless absolutely necessary.
+2.  It is recommended to avoid explicit delays in terms of absolute time to
+    make code emulation friendly. Delays in terms of clock cycles are allowed.
+3.  The `body()` method should only execute the raw functional behavior of
+    the sequence.
+    Any associated housekeeping code should be placed elsewhere, such as in the
+    `pre_start()` and `post_start()` methods.
+    *   `pre_start()` and `post_start()` must be used instead of `pre_body()`
+    and `post_body()`, as these will always be called, even for subsequences
+    called from a parent sequence.
+5.  When creating a sequence object, always randomize the sequence before starting it.
```

Comment:
It does the right thing here, but should probably fix it to `4` in case
you ever break up the list with ```code blocks``` in the future

Reply:
Good catch, fixed